Assets/Scripts/ConnectionManagement/ConnectionState/StartingHostState.cs

```diff
@@ -0,0 +1,110 @@
+using System;
+using Unity.Multiplayer.Samples.BossRoom.Shared.Infrastructure;
+using Unity.Multiplayer.Samples.BossRoom.Shared.Net.UnityServices.Lobbies;
+using Unity.Netcode;
+using Unity.Netcode.Transports.UTP;
+using UnityEngine;
+
+namespace Unity.Multiplayer.Samples.BossRoom
+{
+    /// <summary>
+    /// Connection state corresponding to a host starting up. Starts the host when entering the state. If successful,
+    /// transitions to the Hosting state, if not, transitions back to the Offline state.
+    /// </summary>
+    public class StartingHostState : ConnectionState
+
+    {
+        IPublisher<ConnectStatus> m_ConnectStatusPublisher;
+        LobbyServiceFacade m_LobbyServiceFacade;
+        LocalLobby m_LocalLobby;
+
+        /// <summary>
+        /// How many connections we create a Unity relay allocation for
+        /// </summary>
+        const int k_MaxUnityRelayConnections = 8;
+
+        [Inject]
+        protected void InjectDependencies(IPublisher<ConnectStatus> connectStatusPublisher, LobbyServiceFacade lobbyServiceFacade, LocalLobby localLobby)
+        {
+            m_ConnectStatusPublisher = connectStatusPublisher;
+            m_LobbyServiceFacade = lobbyServiceFacade;
+            m_LocalLobby = localLobby;
+        }
+
+        public override void Enter()
+        {
+            StartHost();
+        }
+
+        public override void Exit() { }
+
+        public override void OnClientDisconnect(ulong clientId)
+        {
+            if (clientId == m_ConnectionManager.NetworkManager.LocalClientId)
+            {
+                m_ConnectStatusPublisher.Publish(ConnectStatus.StartHostFailed);
+                StateChangeRequest?.Invoke(Offline);
+            }
+        }
+
+        public override void OnServerStarted()
+        {
+            m_ConnectStatusPublisher.Publish(ConnectStatus.Success);
+            StateChangeRequest?.Invoke(Hosting);
+        }
+
+        public override void ApprovalCheck(NetworkManager.ConnectionApprovalRequest request, NetworkManager.ConnectionApprovalResponse response)
+        {
+            var connectionData = request.Payload;
+            var clientId = request.ClientNetworkId;
+            // This happens when starting as a host, before the end of the StartHost call. In that case, we simply approve ourselves.
+            if (m_ConnectionManager.NetworkManager.IsHost && clientId == m_ConnectionManager.NetworkManager.LocalClientId)
```

Comment:
Is this check needed, since we're in the "StartingHostState"? Let's make sure we're cleaning up old logic, this current architecture could simplify some of these checks.

Reply:
I see a thumbs up but no changes related to this?